File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s=s.lower()
        l,r=0,len(s)-1
        while l<r:
            while l<r and not((97<=ord(s[l])<=122) or (48<=ord(s[l])<=57)):  
                l+=1
            while l<r and not((97<=ord(s[r])<=122) or (48<=ord(s[r])<=57)):    
                r-=1
            if s[l]!=s[r]:
                return False
            l+=1
            r-=1
        return True
        

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        s = s.toLowerCase();
        StringBuilder filtered = new StringBuilder();
        for(int i=0; i<s.length(); i++){
            if((s.charAt(i) >= 'a' && s.charAt(i) <= 'z') || (s.charAt(i) >= '0' && s.charAt(i) <= '9')){
               filtered.append(s.charAt(i));
            }
        }
        int j=filtered.length()-1;
        for(int i=0; i<filtered.length()/2; i++){
            if(filtered.charAt(i)!=filtered.charAt(j)){
                return false;
            }
            j--;
        }
        return true;
    }
}